File: python/n461_HamDist.py
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        z = x ^ y
        res = 0
        while z > 0:
            if z % 2 != 0:
                res += 1
            z = z // 2
        return res
        
        
if __name__ == "__main__":
    solu = Solution()
    print(solu.hammingDistance(2, 4))

File: cpp/n461_HamDist.cpp
// https://leetcode-cn.com/problems/hamming-distance/submissions/
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int hammingDistance(int x, int y) {
        int z = x ^ y;
        int res = 0;
        while (z > 0){
            if (z % 2 != 0) res += 1;
            z = z / 2;
        }
        return res;
    }
};

int main(){
    Solution solu;
    cout << solu.hammingDistance(2, 4) << endl;
}